provider/datagen/src/transform/cldr/locale_canonicalizer/likely_subtags.rs

```diff
@@ -74,89 +99,156 @@ impl IterableDataProvider<LikelySubtagsForScriptRegionV1Marker> for crate::Datag
-                } else if let Some(region) = entry.0.region {
-                    with_diff!((Language::UND, Some(script), None) => language_region.insert((lang.into(), region.into()), script));
-                } else {
-                    with_diff!((Language::UND, Some(script), Some(region)) => language.insert(lang.into(), (script, region)));
+impl<'a> LikelySubtagsResources<'a> {
+    fn try_from_source_data(
+        source_data: &'a SourceData,
+    ) -> Result<LikelySubtagsResources, DataError> {
+        let likely_subtags: &cldr_serde::likely_subtags::Resource = source_data
+            .cldr()?
+            .core()
+            .read_and_parse("supplemental/likelySubtags.json")?;
+        let coverage_levels: &cldr_serde::coverage_levels::Resource = source_data
+            .cldr()?
+            .core()
+            .read_and_parse("coverageLevels.json")?;
+        let basic_plus_languages = Self::get_basic_plus_languages(coverage_levels);
+        Ok(LikelySubtagsResources {
+            likely_subtags,
+            basic_plus_languages,
+        })
+    }
+
+    fn get_basic_plus_languages(
+        coverage_levels: &cldr_serde::coverage_levels::Resource,
+    ) -> HashSet<Language> {
+        #[allow(clippy::unnecessary_filter_map)] // better for future refactoring
+        coverage_levels
+            .coverage_levels
+            .iter()
+            .filter_map(|(langid, level)| {
+                match level {
+                    // NOTE: If more coverage levels get added, ones below Basic should be filtered here
+                    CoverageLevel::Basic | CoverageLevel::Moderate | CoverageLevel::Modern => {
+                        Some(langid.language)
+                    }
                 }
-            } else if let Some(scr) = entry.0.script {
-                if let Some(region) = entry.0.region {
-                    with_diff!((language, None, None) => script_region.insert((scr.into(), region.into()), language));
+            })
+            .collect()
+    }
+
+    fn get_common(&self) -> LikelySubtagsV1<'static> {
+        transform(
+            self.likely_subtags
+                .supplemental
+                .likely_subtags
+                .iter()
+                .filter(move |(minimized, maximized)| {
+                    self.basic_plus_languages.contains(&maximized.language)
+                        || **minimized == LanguageIdentifier::UND
+                }),
+        )
+    }
+
+    fn get_extended(&self) -> LikelySubtagsV1<'static> {
+        transform(
+            self.likely_subtags
+                .supplemental
+                .likely_subtags
+                .iter()
+                .filter(move |(minimized, maximized)| {
+                    !self.basic_plus_languages.contains(&maximized.language)
+                        || **minimized == LanguageIdentifier::UND
```

Comment:
`nit`: can you implement `get_extended` as `all - get_common` - it's a minor nit but helps with maintainability.